ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py

```diff
@@ -110,20 +112,37 @@ def isFileTest(_file):
 
         self.persistency = persistency
         self.depth = depth
+        self.lfnList = []
         logger.debug("Dataset Created")
 
+    def setReference(self, jobNo, indices):
+        '''A function to set up the references of a subjob's dataset to the master job'''
+        self.ref = jobNo
+        self.files = indices
+
+    def createIndex(self):
+        '''Populate a list of the dataset's LFNs for some quick sorting'''
+        self.lfnList = [df.lfn for df in self.files]
+
     def __getitem__(self, i):
         '''Proivdes scripting (e.g. ds[2] returns the 3rd file) '''
         #this_file = self.files[i]
         # print type(this_file)
         # return this_file
         # return this_file
         # return this_file
+        if self.ref:
+            return GPI.jobs(self.ref).inputdata[self.files][i]
```

Comment:
So is this really the correct way to do this?

Reply:
I couldn't think of a better way...but there probably is one.